client/src/app/site/site.component.scss

```diff
@@ -25,14 +16,51 @@ mat-sidenav-container {
     box-shadow: 3px 0px 10px 0px rgba(0, 0, 0, 0.2);
 }
 
-footer {
+mat-sidenav-container {
+    height: 100%;
+    width: 100%;
     position: fixed;
-    bottom: 0;
+    main {
+        flex: 1;
+        position: relative;
+    }
+}
+
+header {
+    height: 10hv !important;
+}
 
-    span {
-        // width: 100%;
+main {
+    display: flex;
+    flex-direction: column;
+    min-height: 80vh !important;
+    width: 100%;
+    position: relative;
+    flex: 1;
+    > *:not(router-outlet) {
+        flex: 1;
         display: block;
-        text-align: center;
-        margin-bottom: 5px;
+    }
+}
+
+// .page-footer {
+//     min-height: 10vh;
+// }
+.footer-link,
+.footer-right {
+    font-size: 12px;
+}
+
+footer {
+    bottom: 0;
+    width: 100%;
+    height: 10hv !important;
```

Comment:
this will never look clean.
Right now, scrolling starts even if the content could easily fit in range.

I made a little fiddle to give you some inspiration.
http://jsfiddle.net/jx805c4u/

Yes, that's an `absolute` footer and I suppose this is the way to go.

Reply:
the ```absolute``` position causes the footer to appear in mid-text with lazy-loaded content. It has to be ```flex```.

Reply:
that is because the parent is not `relative`.
I will push a patch for you.